core-compose/src/androidTest/java/com/squareup/workflow/ui/compose/internal/ViewFactoriesTest.kt

```diff
@@ -48,14 +60,91 @@ class ViewFactoriesTest {
       WorkflowRendering(TestRendering("two"), viewEnvironment)
     }
 
-    findByText("one\ntwo").assertIsDisplayed()
+    onNodeWithText("one", useUnmergedTree = true).assertIsDisplayed()
+    onNodeWithText("two", useUnmergedTree = true).assertIsDisplayed()
+  }
+
+  @Test fun WorkflowRendering_rendersLegacyAndroidView() {
+    val viewEnvironment = ViewEnvironment(ViewRegistry(LegacyViewViewFactory))
+        .withCompositionRoot { content ->
+          Column {
+            Text("one")
+            content()
+          }
+        }
+
+    composeRule.setContent {
+      WorkflowRendering(LegacyViewRendering("two"), viewEnvironment)
+    }
+
+    onNodeWithText("one", useUnmergedTree = true).assertDoesNotExist()
+    onNodeWithTag(LegacyAndroidViewTestTag, useUnmergedTree = true).assertIsDisplayed()
```

Comment:
Instead of using a test tag here, let's add a dependency on Espresso (to `android-ui-tests.gradle`), and then actually assert what we need to here: `onView(withText("two")).check(matches(isDisplayed()))`. I just verified this locally, and it works (also don't need the `useUnmergedTree` arguments). This makes the tests better (actually testing that the view we care about shown correctly), more readable, and means we don't have to wrap the `emitView` call with something just to pass a test modifier.